extensions/blocks/amazon/edit.js

```diff
@@ -1,23 +1,256 @@
 /**
  * External dependencies
  */
+import tinycolor from 'tinycolor2';
 import { __ } from '@wordpress/i18n';
-import { Component } from '@wordpress/element';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	BlockIcon,
+	ContrastChecker,
+	InspectorControls,
+	PanelColorSettings,
+} from '@wordpress/block-editor';
+import {
+	Button,
+	ExternalLink,
+	FormTokenField,
+	Notice,
+	PanelBody,
+	Placeholder,
+	ToggleControl,
+} from '@wordpress/components';
+import { getBlockDefaultClassName } from '@wordpress/blocks';
+import { useState } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
+import icon from './icon';
+import data from './dummy-data';
 import './editor.scss';
 
-class AmazonEdit extends Component {
-	/**
-	 * Write the block editor UI.
-	 *
-	 * @returns {object} The UI displayed when user edits this block.
-	 */
-	render() {
-		return <p>{ __( 'Coming soon', 'jetpack' ) }</p>;
-	}
-}
+export default function AmazonEdit( {
+	attributes: {
+		backgroundColor,
+		textColor,
+		buttonAndLinkColor,
+		asin,
+		showImage,
+		showTitle,
+		showSeller,
+		showPrice,
+		showPurchaseButton,
+	},
+	className,
```

Comment:
I think you'll need to bring in `name` before you can use it below.

```suggestion
	className,
	name,
```